pkg/providers/vsphere/virtualmachine/snapshot.go

```diff
@@ -218,37 +216,37 @@ func GetSnapshotSize(logger logr.Logger,
 	}
 
 	if snapshot == nil {
-		return 0, fmt.Errorf("snapshot with reference %q"+
+		return 0, 0, fmt.Errorf("snapshot with reference %q"+
 			" not found in vmLayout.Snapshot", vmSnapshot.Value)
 	}
 
-	fcdDeviceKeySet := getFCDDeviceKeySet(moVM)
+	fcdDeviceKeySet, err := getFCDDeviceKeySet(moVM)
+	if err != nil {
+		return 0, 0, err
```

Comment:
Always use the pattern `fmt.Errorf("failed to get FCD device key set: %w", err)` to ensure that the returned error is wrapped with context.